### PR TITLE
TS migration

### PR DESCRIPTION
**Pull request by rohit-gohri**
Adding to PR #1242 as discussed here : https://github.com/graphql/graphiql/pull/1242#issuecomment-580373744

Couldn't open a PR against this branch in graphql repo for some reason. I hope this is okay @acao?

### CODE REVIEW

**Comment by acao** [comment]
@rohit-smpx this is exactly what we want! thank you!

we want it to be additive to ryan’s PR branch which will be the mainline til we merge to master. Ryan himself, as well as I, will be opening some similar PRs against ryan’s branch this weekend and in the coming week as well.

Somewhat similar to a feature branching model? We can all just commit often and stay connected on our Discord channel so we don’t overlap too much, but its not actually a bad thing if we do, because then we might have multiple solutions to evaluate for some of the same code!

thanks! taking a look now

File: packages/graphiql/src/components/ImagePreview.tsx [hunk 1]
```diff
@@ -89,7 +89,7 @@ export class ImagePreview extends React.Component<
           ref={node => {
             this._node = node;
           }}
-          src={tokenToURL(this.props.token)}
+          src={tokenToURL(this.props.token)?.href}
```
Comment (acao) [inline packages/graphiql/src/components/ImagePreview.tsx:92]: now that we’re introducing optional chaining, would you try building the webpack example and making sure it works and loads properly? if not we may want to tweak the example babel configs to include the plugin for this/etc if need be?

Comment (rohit-gohri) [reply packages/graphiql/src/components/ImagePreview.tsx:92]: Will try to build it and change if anything seems to be breaking. Though I don't think it should as [preset-typescript already covers it](https://github.com/babel/babel/issues/10690).

Comment (acao) [reply packages/graphiql/src/components/ImagePreview.tsx:92]: great!

**Comment by rohit-gohri** [comment]
> Somewhat similar to a feature branching model? We can all just commit often and stay connected on our Discord channel so we don’t overlap too much, but its not actually a bad thing if we do, because then we might have multiple solutions to evaluate for some of the same code!

Sounds good. I'll try to fix/improve smaller things now, will attempt more once I understand the project better. Will update on Discord.

File: packages/graphiql/src/components/ToolbarMenu.tsx [hunk 2]
```diff
@@ -75,7 +75,7 @@ export class ToolbarMenu extends React.Component<
     }
   }
 
-  handleClick(e: MouseEvent<HTMLAnchorElement>) {
+  handleClick(e: MouseEvent | React.MouseEvent<HTMLAnchorElement>) {
```
Comment (acao) [inline packages/graphiql/src/components/ToolbarMenu.tsx:78]: ah yes, i remember setting these more specifically when we tried this PR nearly a year ago
I don't think theres a need for a union here, it should always be a React.MouseEvent<> correct?

Comment (rohit-gohri) [reply packages/graphiql/src/components/ToolbarMenu.tsx:78]: These are also being used as native listeners here: https://github.com/rohit-gohri/graphiql/blob/09f4d0b6e6b1c8ccc850d39cfae62c1fd14974b2/packages/graphiql/src/components/ToolbarMenu.tsx#L67

Comment (acao) [reply packages/graphiql/src/components/ToolbarMenu.tsx:78]: ohh, right, good observation

**Pull request merged**